Makefile

```diff
@@ -272,22 +270,21 @@ bin/:
 
 # The build-static is intended to statically link to musl libc.
 # There are possibilities of unexpected errors when statically link to GLIBC.
-build-static: tidy bin/spire-server-static bin/spire-agent-static bin/k8s-workload-registrar-static bin/oidc-discovery-provider-static
+build-static: tidy $(addprefix bin/,$(addsuffix -static,$(binaries)))
 
 # https://7thzero.com/blog/golang-w-sqlite3-docker-scratch-image
 define binary_rule_static
-.PHONY: $1
-$1: | go-check bin/
-	@echo Building $1...
-	$(E)$(go_path) CGO_ENABLED=1 go build $$(go_flags) -ldflags '-s -w -linkmode external -extldflags "-static"' -o $1$(exe) $2
+bin/%: FORCE | go-check
```

Comment:
This rule overwrites the rule brought in by line 254 and causing non-static binaries to be build statically. The macro before this change created unique rule per binary so this problem didn't exist.